pkg/apis/componentconfig/cloudcore/v1alpha1/types.go

```diff
@@ -418,3 +420,19 @@ type Router struct {
 	Port        uint32 `json:"port,omitempty"`
 	RestTimeout uint32 `json:"restTimeout,omitempty"`
 }
+
+// IptablesManager indicates the config of Iptables
+type IptablesManager struct {
+	// Enable indicates whether enable IptablesManager
+	// default true
+	Enable bool `json:"enable"`
+	// It indicates how the component is deployed, valid mode can use "internal" or "external".
+	// The iptables manager component with the internal mode is always deployed inside the cloudcore, will share the host network, forward to the internal port of the tunnel port.
+	// The iptables manager component with the external mode is always deployed outside the cloudcore, will share the host network, forward to the internal cloudcore service and port.
+	// default internal.
+	Mode string `json:"mode,omitempty"`
+	// It indicates the destination port that the component wants to forward.
+	// The stream port(10003) is used by default.
+	// occurs in containerized deployment outside cloudcore, it needs to be modified as needed.
+	ForwardPort uint32 `json:"forwardPort,omitempty"`
```

Comment:
I think we don't need this field in cloudcore? It should always equal to cloudstream's server port.

Reply:
> I think we don't need this field in cloudcore? It should always equal to cloudstream's server port.

Yeah.  Do we acquire the value of cloudstream port when every time the external iptables-manager is set up?  Or dynamically query it when the iptables manger is preparing to set iptables rules.

Reply:
We can store the cloudstream port to configmap? Then external iptables-manager can get it.

Reply:
> We can store the cloudstream port to configmap? Then external iptables-manager can get it.

Yeah. I will dynamically query it when the iptables manger is preparing to set iptables rules.

Reply:
Now hard-code config is enough I think, we can add a task in your issue :)

Reply:
> Now hard-code config is enough I think, we can add a task in your issue :)

Ok.